pingcap/tidb
# topsql: fix issue of topsql failed catch the running SQL when topsql is enabled in execution

### What problem does this PR solve?


Issue Number: close #33859

Problem Summary:

Fix issue of topsql failed catch the running SQL when topsql is enabled in execution

### What is changed and how it works?

Always attach the SQL and Plan information whether TopSQL is enabled.

### Benchmark Test

#### Sysbench test

Machine:  AWS c4.4xlarge, 16 cpu core, 32G memory.
Topology: 1*TiDB, 1*TiKV, 1*PD

Disable TopSQL(threads=300) | QPS in This PR | QPS in Master | Impact(%)
-- | -- | -- | --
Point-Get | 37396 | 37642 | 0.6535253175
Read-Only | 19108 | 19343 | 1.214909786



Enable TopSQL(threads=300) | QPS in This PR | QPS in Master | Impact(%)
-- | -- | -- | --
Point-Get | 34671 | 34643 | -0.08082440897
Read-Only | 17701 | 17682 | -0.1074539079



### Check List

Tests 


1. Create table `t1` and insert into 10000000 rows data.

2. Mock some normal OLTP load by [loadgen](https://github.com/crazycs520/loadgen):

```shell
bin/loadgen payload normal-oltp --rows=1000000
```

3. Execute following SQLs in client-A and client-B at same:

```sql
select count(*),sum(a+b),sum(b+d),sum(c+d*2), sum(d*2+a*b), avg(a*b*d), avg(b*c), avg(c*d),avg(a*b*d), avg(2*b*c), avg(3*c*d), sum(a*b*c), sum(b*d) from t1;
```

```sql
select count(*) from t1 as t0 join t1 as t1 on t0.a != t1.a;
```

4. enable TopSQL during the upper SQL execution:

```sql
set @@global.tidb_enable_top_sql=1;
```

5. Check the TopSQL data.

TiDB: 

![image](https://user-images.githubusercontent.com/26020263/162688286-822ba4ff-39c9-45ae-8aca-d25ddc023bcd.png)

TiKV

![image](https://user-images.githubusercontent.com/26020263/162688436-2b43d410-b7bb-4242-a30f-4c30055332d5.png)


Side effects


### Release note



Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix issue of TopSQL failed to catch the running SQL when TopSQL is enabled in execution.
```
